### PR TITLE
Add Qwen2-VL gpu example

### PR DESCRIPTION
**Pull request by JinheTang**
https://github.com/analytics-zoo/nano/issues/1659

pls kindly review:) @Oscilloscope98

### CODE REVIEW

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    prompt = args.prompt
+
+    # The following code for generation is adapted from https://huggingface.co/Qwen/Qwen2-VL-7B-Instruct#quickstart
+    messages = [
+        {
+            "role": "user",
+            "content": [
+                {
+                    "type": "image",
+                    "image": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen-VL/assets/demo.jpeg",
```
Comment (Oscilloscope98) [inline python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:64]: Please make `image` input as a variable. You could refer to MiniCPM-V-2_6 example: https://github.com/intel-analytics/ipex-llm/tree/main/python/llm/example/GPU/HuggingFace/Multimodal/MiniCPM-V-2_6

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        generated_ids = [
+            output_ids[len(input_ids):] for input_ids, output_ids in zip(inputs.input_ids, generated_ids)
+            ]
+
+        response = processor.batch_decode(generated_ids, skip_special_tokens=True)[0]
+        print(f'Inference time: {end-st} s')
+        print('-'*20, 'Prompt', '-'*20)
+        print(prompt)
+        print('-'*20, 'Output', '-'*20)
+        print(response)
```
Comment (Oscilloscope98) [inline python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:107]: Missing empty line at the end of the file

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+
+    from transformers import Qwen2VLForConditionalGeneration, AutoTokenizer, AutoProcessor
+    from qwen_vl_utils import process_vision_info
+    from ipex_llm import optimize_model
+    model = Qwen2VLForConditionalGeneration.from_pretrained(model_path,
+                                                 trust_remote_code=True,
+                                                 torch_dtype = 'auto',
+                                                 use_cache=True)
+    model = optimize_model(model, low_bit='sym_int4')
```
Comment (Oscilloscope98) [inline python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:47]: Please refer to [here](https://github.com/intel-analytics/ipex-llm/blob/main/python/llm/example/GPU/PyTorch-Models/Model/llama3/generate.py#L57) to add several more augments :)

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py (hunk 1 above)
Comment (Oscilloscope98) [reply python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:64]: Also updated the sample image based on MiniCPM-V-2_6 example

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import time
+import argparse
+
+from transformers import AutoTokenizer
+from ipex_llm import optimize_model
+import numpy as np
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Qwen2-VL-7B-Instruct')
```
Comment (Oscilloscope98) [inline python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:27]: 'Predict Tokens using `generate()` API for Qwen2-VL-7B-Instruct

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    from transformers import Qwen2VLForConditionalGeneration, AutoTokenizer, AutoProcessor
+    from qwen_vl_utils import process_vision_info
+    from ipex_llm import optimize_model
+    model = Qwen2VLForConditionalGeneration.from_pretrained(model_path,
+                                                 trust_remote_code=True,
+                                                 torch_dtype = 'auto',
+                                                 use_cache=True)
+    model = optimize_model(model, low_bit='sym_int4')
+    model = model.to("xpu")
```
Comment (Oscilloscope98) [inline python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:48]: Add information regarding:

`model = model.float().eval()    # use .float() for better output, use .half() for better speed`

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+    # Load processor
+    min_pixels = 256*28*28
+    max_pixels = 1280*28*28
+    processor = AutoProcessor.from_pretrained(model_path, min_pixels=min_pixels, max_pixels=max_pixels)
```
Comment (Oscilloscope98) [inline python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:53]: Update for why we need `min_pixels`/`max_pixels`

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py (hunk 3 above)
Comment (JinheTang) [reply python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:51]: `low_cpu_mem_usage=True` doesn't seem to apply to `Qwen2VLForConditionalGeneration.from_pretrained`:
image
error persists after installation of `accelerate`

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py (hunk 5 above)
Comment (JinheTang) [reply python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py]: issue launched: https://github.com/analytics-zoo/nano/issues/1659#issuecomment-2381115065

**Comment by Oscilloscope98** [comment]
Missing updating the table [here](https://github.com/intel-analytics/ipex-llm?tab=readme-ov-file#verified-models) and [here](https://github.com/intel-analytics/ipex-llm/blob/main/README.zh-CN.md)

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    args = parser.parse_args()
+    model_path = args.repo_id_or_model_path
+
+
+    from transformers import Qwen2VLForConditionalGeneration, AutoTokenizer, AutoProcessor
+    from qwen_vl_utils import process_vision_info
+    from ipex_llm import optimize_model
+    model = Qwen2VLForConditionalGeneration.from_pretrained(model_path,
+                                                 trust_remote_code=True,
+                                                 torch_dtype = 'auto',
```
Comment (Oscilloscope98) [inline python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:49]: `torch_dtype='auto'`

File: python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+    # The default range for the number of visual tokens per image in the model is 4-16384. You can set min_pixels and max_pixels according to your needs, such as a token count range of 256-1280, to balance speed and memory usage.
+    min_pixels = 256*28*28
+    max_pixels = 1280*28*28
+    processor = AutoProcessor.from_pretrained(model_path, min_pixels=min_pixels, max_pixels=max_pixels)
+
+    prompt = args.prompt
+    image_path = args.image_url_or_path
+
+    # The following code for generation is adapted from https://huggingface.co/Qwen/Qwen2-VL-7B-Instruct#quickstart
```
Comment (Oscilloscope98) [inline python/llm/example/GPU/PyTorch-Models/Model/qwen2-vl/generate.py:63]: Let's move this comment after `model = model.float().to("xpu")`, as code regrading processor is also referred from Qwen2-VL-7B-Instruct's quickstart

**Review by Oscilloscope98 — APPROVED** [review]

**Pull request merged**